### PR TITLE
docs: add simple introduction and design goals to README

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR updates main README.md for devconnect event. There will be a couple of visitors during the event and we want to clarify what it is for.

### CODE REVIEW

**Review by Insun35 — APPROVED** [review]
LGTM

**Review by dongchangYoo — APPROVED** [review]

**Comment by dongchangYoo** [comment]
LGTM

File: README.md [hunk 1]
```diff
@@ -4,7 +4,16 @@
...
 </div>
 
-Tachyon is a ZKP accelerator using CUDA.
+# Overview
+
+**Tachyon** is a Modular ZK Backend, powered by GPU.
+
+## Design Goals
+
+1. General Purpose: We want a code that enables switch different proving scheme.
```
Comment (seohoyoon93) [inline README.md:13]: 1. General Purpose: A versatile ZK library empowers developers to implement any proving scheme with minimal effort, often enhancing developer productivity. To create a general-purpose backend, aligning the code structure as closely as possible with the algebraic structure is paramount.

File: README.md [hunk 2]
```diff
@@ -4,7 +4,16 @@
...
+2. Easy to use: Switching code should be easy.
```
Comment (seohoyoon93) [inline README.md:14]: 2. Easy to Use: Achieving widespread adoption is essential for the success of any product. Consequently, one of the key focal points of the Tachyon project is to include offering packages for various programming languages and runtimes.

File: README.md [hunk 3]
```diff
@@ -4,7 +4,16 @@
...
+3. Blazing Fast: Of course, it should be fast.
```
Comment (seohoyoon93) [inline README.md:15]: 3. Blazing Fast: Tachyon's foremost requirement is speed, and not just any speed, but blazing speed! This entails Tachyon delivering exceptional performance on both CPU and GPU platforms.

File: README.md [hunk 4]
```diff
@@ -4,7 +4,16 @@
...
+4. GPU Interoperability: We want to write a code once and have it run on both CPU and GPU as many as possible.
```
Comment (seohoyoon93) [inline README.md:16]: 4. GPU Interoperability: Tachyon's code is designed to be compatible with both CPU and GPU in the majority of scenarios.

**Pull request merged**